src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -53,6 +60,9 @@
     QuotedSemiStructuredElementSegment=NamedSegment.make(
         'double_quote', name='quoted_semi_structured_element',
         type='semi_structured_element'),
+    ColumnIndexIdentifierSegment=ReSegment.make(
+        r"\$[0-9]+", name='column_index_identifier_segment',
+        type='column_index_identifier'),
```

Comment:
Is the type name here just arbitrary @alanmcruickshank ?

Reply:
In this case that's an excellent question. Both `type` and `name` can be referred to downstream by linting rules when linting. In this case I think this should be of `type` `identifier` because I think several rules should also apply to these. It's possible that it might cause issues but I'd like to at least try that. The `name` you've chosen looks sensible.